src/software/ai/hl/stp/play/offense/offense_play_test.py

```diff
@@ -67,18 +69,21 @@ def test_offense_play(simulated_test_runner):
     )
 
     # Always Validation
-    # TODO- #2779 Validation
-    always_validation_sequence_set = [[]]
+    inv_always_validation_sequence_set = [
+        [BallAlwaysStaysInRegion(regions=[field.fieldBoundary()])]
+    ]
+
+    ag_always_validation_sequence_set = [[FriendlyAlwaysHasBallPossession()]]
```

Comment:
nit suggestion: you can re-order the initialization of the variable arguments so that they follow the same order as the list of method parameters, when they are passed to the test runner below. This helps with readability and consistency.